### DISCUSSION TITLE
Derivatives of Dynamics implmented in Pinocchio

### DISCUSSION THREAD

**Discussion by Yingfan99327** [stack-of-tasks/pinocchio#2432] [General]
Just a small question that whether Pinocchio could be used to compute the analytical derivatives of the RNEA/ABA algorithms w.r.t the inertia parameters？

**Comment by cmastalli** [comment]
In our recent work, you can see how to compute these analytical derivatives while ensuring full physical consistency: https://arxiv.org/pdf/2403.17161v1

We will release this algorithm, together with other useful things, in Crocoddyl.

However, we could consider to implement these analytical derivatives in Pinocchio.

**Reply by Yingfan99327** [reply]
That's really great! Thanks so much!!!

After reading your paper, especially about the parameterization of inertia parameters section, may I ask how to calculate the the Y matrix in the (19) equation, I know such an affine relationship in robotic manipulator, however, I guess directly calculating such a matrix probably is very difficult and complex of a 6-dof or 7-dof manipulator, maybe? I guess the analytical expression should be a combination of multiple cos and sin function?

Sorry for my follow-up, because I really want to know the "exact" expression of the "analytical" expression of the derivatives of the inverse dynamics w.r.t the inertia parameters.

Thanks very much!!!

**Reply by Yingfan99327** [reply]
Will the regressor matrix automatically calculated by the func provided in Pinocchio, for instance, computeJointTorqueRegressor?

**Reply by cmastalli** [reply]
$\mathbf{Y(q, v, \dot{v})}$ denotes the joint-torque regressor matrix and we compute it using Pinocchio.

**Reply by Yingfan99327** [reply]
Thanks very much for your reply!
